All_home_works/hw2/task3.py

```diff
@@ -0,0 +1,35 @@
+"""
+Write a function that takes K lists as arguments and returns all possible
+lists of K items where the first element is from the first list,
+the second is from the second and so one.
+
+You may assume that that every list contain at least one element
+
+Example:
+
+assert combinations([1, 2], [3, 4]) == [
+    [1, 3],
+    [1, 4],
+    [2, 3],
+    [2, 4],
+]
+"""
+from typing import Any, List
+
+
+def combinations(*args: List[Any]) -> List[List]:
+    res, nums = [], []
+    if args:
+        recursion(res, nums, args)
+    return res
+
+
+def recursion(res, nums, args, index=0):
```

Comment:
Could we structure this program in a way when you avoid passing `nums` as an argument?

Reply:
No problem!